prober/icmp.go

```diff
@@ -58,40 +59,44 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		return false
 	}
 
+	var srcIP net.IP
+	if len(module.ICMP.SourceIPAddress) > 0 {
+		if srcIP = net.ParseIP(module.ICMP.SourceIPAddress); srcIP == nil {
+			level.Error(logger).Log("msg", "Error parsing source ip address", "srcIP", module.ICMP.SourceIPAddress)
+			return false
+		}
+		level.Info(logger).Log("msg", "Using source address:", "srcIP", srcIP)
+	}
+
 	level.Info(logger).Log("msg", "Creating socket")
 	if ip.IP.To4() == nil {
 		requestType = ipv6.ICMPTypeEchoRequest
 		replyType = ipv6.ICMPTypeEchoReply
 
-		socket, err = icmp.ListenPacket("ip6:ipv6-icmp", "::")
+		icmpConn, err := icmp.ListenPacket("ip6:ipv6-icmp", "::")
 		if err != nil {
 			level.Error(logger).Log("msg", "Error listening to socket", "err", err)
 			return
 		}
+
+		socket = icmpConn
+		socket6 = icmpConn.IPv6PacketConn()
 	} else {
 		requestType = ipv4.ICMPTypeEcho
 		replyType = ipv4.ICMPTypeEchoReply
 
-		if !module.ICMP.DontFragment {
-			socket, err = icmp.ListenPacket("ip4:icmp", "0.0.0.0")
-			if err != nil {
-				level.Error(logger).Log("msg", "Error listening to socket", "err", err)
-				return
-			}
-		} else {
-			s, err := net.ListenPacket("ip4:icmp", "0.0.0.0")
-			if err != nil {
-				level.Error(logger).Log("msg", "Error listening to socket", "err", err)
-				return
-			}
+		s, err := net.ListenPacket("ip4:icmp", "0.0.0.0")
+		if err != nil {
+			level.Error(logger).Log("msg", "Error listening to socket:", "err", err)
+			return
```

Comment:
Explicitly returning false would be clearer.

Reply:
I agree. That is not my code. I just moved it. I am fine with doing a cleanup commit but would separate it from this PR.